### PR TITLE
FFM-9086: Add a terraform resource to create a new target group

### PR DESCRIPTION
**Pull request by ribeirophillipe**
## Describe your changes
* Add resource feature_flag_target_group
* Add unit tests
* 
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,4 +29,4 @@ test:
 
 sweep:
 	@echo "WARNING: This will destroy infrastructure. Use only in development accounts."
-	go test $(SWEEP_DIR) -v -sweep=all $(SWEEPARGS) -timeout 60m
+	go test $(SWEEP_DIR) -v -sweep=all $(SWEEPARGS) -timeout 60m
```
Comment (davejohnston) [inline Makefile:32]: Think this is just whitespace.

File: examples/resources/harness_platform_slo/resource.tf [hunk 2]
```diff
@@ -44,4 +44,4 @@ resource "harness_platform_slo" "example" {
       enabled = true
     }
   }
-}
+}
```
Comment (davejohnston) [inline examples/resources/harness_platform_slo/resource.tf:47]: more whitespace

File: internal/provider/provider.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/harness/terraform-provider-harness/internal/service/platform/feature_flag"
 	"github.com/harness/terraform-provider-harness/internal/service/platform/feature_flag_target"
+	featureflagtargetgroup "github.com/harness/terraform-provider-harness/internal/service/platform/feature_flag_target_group"
```
Comment (davejohnston) [inline internal/provider/provider.go:10]: do we need to alias this?

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Description: "The name of the feature flag target group.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"included": {
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:64]: I would phrase this as "A list of targets to to include in the target group"

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+				Description: "The list of rules to include in the feature flag target group.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				MinItems:    0,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"excluded": {
+				Description: "The list of rules to include in the feature flag target group.",
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:74]: typo here - it says include but should be exclude.  Similar to above, I'd state its a list of 'targets' to be excluded from the target group

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+				Type:        schema.TypeList,
+				Optional:    true,
+				MinItems:    0,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"rules": {
+				Description: "The list of rules to exclude in the feature flag target group.",
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:83]: This should be 'include' not exclude

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"attribute": {
+							Description: "The attribute to use in the clause.  This can be any target attribute",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"id": {
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:93]: I don't think we should expose this 'id'.  It should be generated automatically on the server-side.

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+		d.MarkNewResource()
+	}
+
+	httpResp, err = c.TargetGroupsApi.CreateSegment(ctx, segmentRequest, c.AccountId, qp.OrgID)
+
+	if err != nil {
+		return helpers.HandleApiError(err, d, httpResp)
+	}
+
+	time.Sleep(1 * time.Second)
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:195]: Do we need this sleep?  If we check the status code from httpResp that might be more reliable.

**Review by davejohnston — COMMENTED** [review]
Overall looks good.
There is some whitespace changes to files that can be reverted.
Take a look at the descriptions fro the resources, I think these  need some changes.
One query over the sleep - do we need to use it, could we check the resp code from the create operation?

File: internal/provider/provider.go (hunk 3 above)
Comment (ribeirophillipe) [reply internal/provider/provider.go:10]: compiler complained without one ...

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go (hunk 7 above)
Comment (ribeirophillipe) [reply internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:93]: that was based on the original schema... I will remove it.

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go (hunk 8 above)
Comment (ribeirophillipe) [reply internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:195]: done

File: internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+				Description: "A list of targets to exclude from the target group",
+				Type:        schema.TypeList,
+				Optional:    true,
+				MinItems:    0,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"rules": {
+				Description: "The list of rules to include in the feature flag target group.",
```
Comment (davejohnston) [inline internal/service/platform/feature_flag_target_group/resource_feature_flag_target_group.go:83]:
```suggestion
				Description: "The list of rules used to include targets in the target group.",
```

**Review by davejohnston — DISMISSED** [review]
Please make one last change to the rules description.
```The list of rules used to include targets in the target group```

File: internal/service/cd/delegate/delegate_helper_test.go [hunk 10]
```diff
@@ -63,7 +64,7 @@ func deleteDelegate(t *testing.T, name string) {
 	cli, err := client.NewClientWithOpts(client.FromEnv)
 	require.NoError(t, err, "failed to create docker client: %s", err)
 
-	err = cli.ContainerStop(context.Background(), name, nil)
+	err = cli.ContainerStop(context.Background(), name, container.StopOptions{})
```
Comment (rathodmeetsatish) [inline internal/service/cd/delegate/delegate_helper_test.go:67]: remove changes not related to ff

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
generate docs

**Comment by rathodmeetsatish** [comment]
trigger build

**Review by rathodmeetsatish — APPROVED** [review]

**Review by davejohnston — APPROVED** [review]
Looks good

**Pull request merged**